src/test/groovy/org/prebid/server/functional/tests/module/richmedia/RichMediaFilterSpec.groovy

```diff
@@ -0,0 +1,340 @@
+
+        then: "Response header should contain seatbid"
+        assert response.seatbid.size() == 1
+
+        and: "Response shouldn't contain errors of invalid creation"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain analytics"
+        assert !getAnalyticResults(response)
+
+        where:
+        amdValue << [PATTERN_NAME, "${PBSUtils.randomString}-${PATTERN_NAME}", "${PATTERN_NAME}-${PBSUtils.randomString}"]
+    }
+
+    def "PBS should prioritize account config and reject request with error and provide analytic when adm matches with pattern name on filters in account config and host config is enabled"() {
+        given: "BidRequest with stored response"
+        def storedResponseId = PBSUtils.randomNumber
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            it.ext.prebid.trace = VERBOSE
+            it.imp.first().ext.prebid.storedBidResponse = [new StoredBidResponse(id: storedResponseId, bidder: GENERIC)]
+        }
+
+        and: "Account with enabled richMedia config in the DB"
+        def richMediaFilterConfig = new PbsModulesConfig(pbRichmediaFilter: new RichmediaFilter(filterMraid: true, mraidScriptPattern: PATTERN_NAME_ACCOUNT))
+        def accountConfig = new AccountConfig(hooks: new AccountHooksConfiguration(modules: richMediaFilterConfig))
+        def account = new Account(uuid: bidRequest.getAccountId(), config: accountConfig)
+        accountDao.save(account)
+
+        and: "Stored bid response in DB"
+        def storedBidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            it.seatbid[0].bid[0].adm = PATTERN_NAME_ACCOUNT
+        }
+        def storedResponse = new StoredResponse(responseId: storedResponseId, storedBidResponse: storedBidResponse)
+        storedResponseDao.save(storedResponse)
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithDisabledMediaFilter.sendAuctionRequest(bidRequest)
+
+        then: "Response header shouldn't contain any seatbid"
+        assert !response.seatbid
+
+        and: "Response should contain error of invalid creation for imp with code 350"
+        def responseErrors = response.ext.errors
+        assert responseErrors[ErrorType.GENERIC]*.message == ['Invalid creatives']
+        assert responseErrors[ErrorType.GENERIC]*.code == [350]
+        assert responseErrors[ErrorType.GENERIC].collectMany { it.impIds } == bidRequest.imp.id
+
+        and: "Add an entry to the analytics tag for this rejected bid response"
+        def analyticsTags = getAnalyticResults(response)
+        assert analyticsTags.size() == 1
+        def analyticResult = analyticsTags.first()
+        assert analyticResult == AnalyticResult.buildFromImp(bidRequest.imp.first())
+    }
+
+    def "PBS should prioritize account config and process request without analytics when adm matches with pattern name on filters in host config and account config is enabled"() {
```

Comment:
At this test you can't know who will be prioritize when host config disabled

Reply:
updated

Reply:
imho test name is a bit vague for me, maybe makes sense to mention that the pattern matches with the host pattern while both configs are enabled